src/core/metrics.cc

```diff
@@ -368,11 +426,48 @@ void impl::add_registration(const metric_id& id, const metric_type& type, metric
         _value_map[name].info().inherit_type = type.type_name;
         _value_map[name].info().name = id.full_name();
         _value_map[name].info().aggregate_labels = aggregate_labels;
-        _value_map[name][id.labels()] = rm;
+        _value_map[name][rm->info().id.labels()] = rm;
     }
     dirty();
 }
 
+void impl::set_relabel_configs(std::vector<relabel_config> relabel_configs) {
+    _relabel_configs = std::move(relabel_configs);
+    for (auto&& family : _value_map) {
+        std::vector<shared_ptr<registered_metric>> rms;
```

Comment:
Let's say I set relabel_configs, then drop it. Any altered/removed metrics won't come back.

So I think you need to keep the original metrics, and re-apply the label configuration when it changes.

Reply:
I wasn't sure about it, naturally, keeping the original label is better, note that when you drop a metric it doesn't get deleted, it is simply marked disable.
There is a problem with labels, so I can keep the original set of labels, there is a question if there will be an external reset all, or if the calculation will start always from the original labels

Reply:
I think you should always apply the filter to the original.